### PR TITLE
Implement Private Set Intersection example

### PR DESCRIPTION
**Pull request by tiziano88**
ref. #31

### CODE REVIEW

File: examples/private_set_intersection/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Private Set Intersection example for Project Oak.
+//!
+//! Clients invoke the module by providing their own private set, and the module keeps track of the
+//! intersection of all the provided sets from all the clients that have interacted with it.
```
Comment (michael-kernel-sanders) [inline examples/private_set_intersection/src/lib.rs]: What does it do with the intersection?

File: examples/private_set_intersection/client/private_set_intersection.cc [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  ::google::protobuf::Empty response;
+  ::grpc::Status status = stub->SubmitSet(&context, request, &response);
+  if (!status.ok()) {
+    LOG(QFATAL) << "Could not submit set: " << status.error_code() << ": "
+                << status.error_message();
+  }
+}
+
+std::vector<std::string> retrieve_intersection(PrivateSetIntersection::Stub* stub) {
```
Comment (michael-kernel-sanders) [inline examples/private_set_intersection/client/private_set_intersection.cc]: CamelCase function names

File: examples/private_set_intersection/src/lib.rs (hunk 1 above)
Comment (tiziano88) [reply examples/private_set_intersection/src/lib.rs]: Clarified,

File: examples/private_set_intersection/client/private_set_intersection.cc (hunk 2 above)
Comment (tiziano88) [reply examples/private_set_intersection/client/private_set_intersection.cc]: Done.

**Pull request merged**